utils/polus-bfio-util/bfio/bfio.py

```diff
@@ -1,457 +1,210 @@
-        else:
-            rdr = jutil.JClassWrapper('loci.formats.ImageReader')()
-        rdr.setOriginalMetadataPopulated(True)
+        Note:
+            Not all methods of indexing can be used, and some indexing will lead
+            to unexpected results. For example, logical indexing cannot be used,
+            and step sizes in slice objects is ignored for the first three
+            indices. This means and index such as ``[0:100:2,0:100:2,0,0,0]``
+            will return a 100x100x1x1x1 numpy array.
         
-        # Access the OMEXML Service
-        clsOMEXMLService = jutil.JClassWrapper(
-            'loci.formats.services.OMEXMLService')
-        serviceFactory = jutil.JClassWrapper(
-            'loci.common.services.ServiceFactory')()
-        service = serviceFactory.getInstance(clsOMEXMLService.klass)
-        omexml = service.createOMEXMLMetadata()
+        Parameters:
+            keys: numpy-like slicing used to load a section of an image.
         
-        # Read the metadata
-        rdr.setMetadataStore(omexml)
-        rdr.setId(self._file_path)
-
-        # Parse it using the OMEXML class
-        self._metadata = bioformats.OMEXML(omexml.dumpXML())
-        return self._metadata
-
-    def _val_xyz(self, xyz, axis):
-        """_val_xyz Utility function for validating image dimensions
-
-        Args:
-            xyz (int): Pixel value of x, y, or z dimension.
-                If None, returns the maximum range of the dimension
-            axis (str): Must be 'x', 'y', or 'z'
-
         Returns:
-            list: list of ints indicating the first and last index in the dimension
-        """
-        assert axis in 'XYZ'
-        if not xyz:
-            xyz = [0, self._xyzct[axis]]
-        else:
-            assert len(xyz) == 2,\
-                '{} must be a list or tuple of length 2.'.format(axis)
-            assert xyz[0] >= 0,\
-                '{}[0] must be greater than or equal to 0.'.format(axis)
-            assert xyz[1] <= self._xyzct[axis],\
-                '{}[1] cannot be greater than the maximum of the dimension ({}).'.format(axis, self._xyzct[axis])
-        return xyz
-
-    def _val_ct(self, ct, axis):
-        """_val_ct Utility function for validating image dimensions
+            A numpy.ndarray where trailing empty dimensions are removed.
+            
+        Example:
 
-        Args:
-            ct (int,list): List of ints indicating the channels or timepoints to load
-                If None, returns a list of ints
-            axis (str): Must be 'c', 't'
+            .. code-block:: python
 
-        Returns:
-            list: list of ints indicating the first and last index in the dimension
+                import bfio
+                
+                # Initialize the bioreader
+                br = bfio.BioReader('Path/To/File.ome.tif')
+                
+                # Load and copy a 100x100 array of pixels
+                a = br[:100,:100,0,0,0]
```

Comment:
Correction: `a = br[:100,:100,0:1,0,0]` because index in position 2 should be a slice